plugin/src/main/java/at/helpch/chatchat/data/impl/gson/GsonDatabase.java

```diff
@@ -0,0 +1,174 @@
+package at.helpch.chatchat.data.impl.gson;
+
+import at.helpch.chatchat.ChatChatPlugin;
+import at.helpch.chatchat.api.ChatUser;
+import at.helpch.chatchat.data.base.Database;
+import at.helpch.chatchat.user.ChatUserImpl;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParseException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.UUID;
+import java.util.logging.Level;
+import org.jetbrains.annotations.NotNull;
+
+public class GsonDatabase implements Database {
+    private final ChatChatPlugin plugin;
+    private final Gson gson;
+    private final File usersDirectory;
+
+    public GsonDatabase(@NotNull final ChatChatPlugin plugin) {
+        this.plugin = plugin;
+        this.gson = new GsonBuilder()
+            .registerTypeAdapter(ChatUser.class, new ChatUserAdapterFactory(plugin))
+            .registerTypeAdapter(ChatUserImpl.class, new ChatUserAdapterFactory(plugin))
```

Comment:
Should this just have one instance of the `ChatUserAdapterFactory`

In my bit of testing, I had one instance that I passed to both and it was working fine.